### ISSUE TITLE
Empty error description when deploying a smart contract (Detailed trace)

### ISSUE THREAD

**Issue by SmartArray** [trufflesuite/truffle#5297]
## Issue
When deploying a smart contract, sometimes we get a stack trace like
```
Error: 
    at /usr/local/lib/node_modules/truffle/build/webpack:/packages/deployer/src/deployment.js:331:1
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Migration._deploy (/usr/local/lib/node_modules/truffle/build/webpack:/packages/migrate/Migration.js:69:1)
    at Migration._load (/usr/local/lib/node_modules/truffle/build/webpack:/packages/migrate/Migration.js:55:1)
    at Migration.run (/usr/local/lib/node_modules/truffle/build/webpack:/packages/migrate/Migration.js:203:1)
    at Object.runMigrations (/usr/local/lib/node_modules/truffle/build/webpack:/packages/migrate/index.js:143:1)
    at Object.runFrom (/usr/local/lib/node_modules/truffle/build/webpack:/packages/migrate/index.js:108:4)
    at Object.runAll (/usr/local/lib/node_modules/truffle/build/webpack:/packages/migrate/index.js:112:4)
    at Object.run (/usr/local/lib/node_modules/truffle/build/webpack:/packages/migrate/index.js:85:1)
    at Object.run (/usr/local/lib/node_modules/truffle/build/webpack:/packages/core/lib/testing/Test.js:114:1)
    at startGanacheAndRunTests (/usr/local/lib/node_modules/truffle/build/webpack:/packages/core/lib/commands/test/run.js:49:1)
    at Object.module.exports [as run] (/usr/local/lib/node_modules/truffle/build/webpack:/packages/core/lib/commands/test/run.js:106:1)
    at runCommand (/usr/local/lib/node_modules/truffle/build/webpack:/packages/core/lib/command-utils.js:184:1)
```

This is not very developer friendly and should be changed.

Fortunately, I was able to dig into the webpack-compiled bundles and take a look what's going on.

https://github.com/trufflesuite/truffle/blob/b9ed05e806b5de524cc1c5489cdd28e0fced573f/packages/deployer/src/sanitizeMessage.js#L6-L7

just tries to print the first element of the error array. By utilizing the environment variable ```DEBUG=deploy*```, I got 
```bash
deployer:sanitizeMessage processing the following message - [ undefined, [], [], [], [], [], [], [] ] +0ms
```

This was of course not very helpful either. But by logging the error that happens here:

https://github.com/trufflesuite/truffle/blob/b9ed05e806b5de524cc1c5489cdd28e0fced573f/packages/deployer/src/deployment.js#L328

gave me a hint that the error that was happening on our site was indeed very simple:
```
Error: overflow [ See: https://links.ethers.org/v5-errors-NUMERIC_FAULT-overflow ] (fault="overflow", operation="BigNumber.from", value=1000000000000000000, code=NUMERIC_FAULT, version=bignumber/5.6.2)
```

The bad error handling costed me half an hour for such a simple thing, and an additional half for writing this issue, when in fact it could be simpler 😱

## Trace

This causes a major problem:
https://github.com/trufflesuite/truffle/blob/ff5e098f75fce170de84ad0aac7ab5a94abc2ffe/packages/events/defaultSubscribers/migrate/index.js#L86

Although I didn't set it to quiet explicitly, it will return `undefined` by default so that the first element of the array is undefined and `sanitizeMessage` won't print out anything.

Wouldn't it be better to pass the error object or message to `sanitizeMessage`?
It would at least contain a valid error message...

## Steps to Reproduce
Deploy a contract with an invalid big integer (e.g. 10**18).

## Expected Behavior
Better error message handling, for example: Error: NUMERIC_FAULT or Error: overflow

## Actual Results
No detailed error description

## Environment
```
MacOS
Truffle v5.5.21 (core: 5.5.21)
Ganache v7.2.0
Solidity - ^0.8.15 (solc-js)
Node v16.15.1
Web3.js v1.7.4
```

**Comment by haltman-at** [comment]
Hm, when I try this I'm seeing the error message just fine.  Could you give more detailed reproduction steps?  Thanks!

**Comment by mipicdev** [comment]
I had the same issue when executing `truffle test --network=test`.

This error occurred because I had forgotten to add the condition `network === "test"` to the corresponding migration file:

```
const MyERC777Token = artifacts.require("MyERC777Token");
const { singletons } = require("@openzeppelin/test-helpers");
require("@openzeppelin/test-helpers/configure")({ provider: web3.currentProvider, environment: "truffle" });

module.exports = async function (deployer, network, accounts) {
  if (network === "development" || network === "test") {
    await singletons.ERC1820Registry(accounts[0]);
  }

  await deployer.deploy(MyERC777Token);
};

```

**Comment by cds-amal** [comment]
@SmartArray, closing as this should be fixed in [Truffle v5.5.24](https://github.com/trufflesuite/truffle/releases/tag/v5.5.24). Kindly let me know if it isn't.
